### PR TITLE
IndirectFittingModel Unit tests

### PR DESCRIPTION
**Pull request by robertapplin**
**Merge #23516 first**

**Description of work.**
This PR introduces unit tests for IndirectFittingModel.

Part of #23463

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/scientific_interfaces/Indirect/IndirectFittingModel.cpp [hunk 1]
```diff
@@ -508,7 +510,8 @@ void IndirectFittingModel::addNewWorkspace(MatrixWorkspace_sptr workspace,
 }
 
 void IndirectFittingModel::removeWorkspace(std::size_t index) {
-  removeFittingData(index);
+  if (m_fittingData.size() > index)
```
Comment (TWJubb) [inline qt/scientific_interfaces/Indirect/IndirectFittingModel.cpp]: Should this if statement also check for non-negative index?

Comment (robertapplin) [reply qt/scientific_interfaces/Indirect/IndirectFittingModel.cpp]: The index will always be positive or zero because it is a `std::size_t`

**Review by TWJubb — DISMISSED** [review]
Very neat suite of tests; same as the last one. 

- Short tests.
- Good names.
- Well structured.
- Very readable.

Very nice. 

Only one suggestion; there are quite a few tests, is there a way to structure them a bit more? The first thing is to order them in a sensible way, you could also add some comments to help section the tests, for example using

```
// --------------------------------
// Test Section 1
// --------------------------------
```

This just helps the tests become a little more readable; as associated functionality is tested in each section.

This looks great though so I am approving without changes; but be prepared as you add tests on dependent code to add to these tests.

File: qt/scientific_interfaces/Indirect/IndirectFittingModel.cpp [hunk 2]
```diff
@@ -412,7 +412,9 @@ std::size_t IndirectFittingModel::numberOfWorkspaces() const {
 }
 
 std::size_t IndirectFittingModel::getNumberOfSpectra(std::size_t index) const {
-  return m_fittingData[index]->numberOfSpectra();
+  if (index < m_fittingData.size())
+    return m_fittingData[index]->numberOfSpectra();
+  return 0;
```
Comment (quantumsteve) [inline qt/scientific_interfaces/Indirect/IndirectFittingModel.cpp]: If one gives this function an invalid index, should the user be aware that this happened? Perhaps one should log a warning and/or throw an exception?

File: qt/scientific_interfaces/Indirect/IndirectFittingModel.cpp (hunk 1 above)
Comment (quantumsteve) [reply qt/scientific_interfaces/Indirect/IndirectFittingModel.cpp]: If one gives this function an invalid index, should the user be aware that this happened? Perhaps one should log a warning and/or throw an exception?

**Review by quantumsteve — CHANGES_REQUESTED** [review]
Should we give users some feedback about out-of-bounds indices?

**Review by quantumsteve — APPROVED** [review]

**Pull request merged**